ios/Components/SessionHelperModule.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+import Adyen
+
+protocol SessionResultListener {
+    func didComplete(with result: Adyen.AdyenSessionResult)
+    func didFail(with error: Error)
+}
+
+@objc(SessionHelper)
+internal final class SessionHelperModule: BaseModule, AdyenSessionDelegate {
+
+    internal static var sessionListener: SessionResultListener?
+
+    func didComplete(with result: Adyen.AdyenSessionResult, component: Adyen.Component, session: Adyen.AdyenSession) {
+        SessionHelperModule.sessionListener?.didComplete(with: result)
+    }
+
+    func didFail(with error: Error, from component: Adyen.Component, session: Adyen.AdyenSession) {
+        SessionHelperModule.sessionListener?.didFail(with: error)
+    }
+
+    func didOpenExternalApplication(component: Adyen.ActionComponent, session: Adyen.AdyenSession) {
+    }
+
+    override public func supportedEvents() -> [String]! { [Events.didComplete.rawValue, Events.didFail.rawValue] }
+
+
+    @objc
+    func createSession(_ sessionModelJSON: NSDictionary,
+                       configuration: NSDictionary,
+                       resolver: @escaping RCTPromiseResolveBlock,
+                       rejecter: @escaping RCTPromiseRejectBlock) {
+        let parser = RootConfigurationParser(configuration: configuration)
+        let context: AdyenContext
+        do {
+            context = try parser.fetchContext()
+        } catch {
+            return rejecter("session", nil, error)
+        }
+
+        guard let id = sessionModelJSON["id"] as? String, let data = sessionModelJSON["sessionData"] as? String else {
+            return rejecter("session", "Invalid session data", nil)
+        }
+
+        let config = AdyenSession.Configuration(sessionIdentifier: id, initialSessionData: data, context: context)
+        DispatchQueue.main.async {
```

Comment:
Is this super necessary?

Reply:
yep 🤷‍♂️
React Native calling native modules in non-UI thread